### PR TITLE
contextifying pass through proof checkers

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: go/engine/identify2_test.go [hunk 1]
```diff
@@ -71,11 +71,11 @@ func (i *Identify2WithUIDTester) MakeProofChecker(_ libkb.RemoteProofChainLink)
 	return i, nil
 }
 
-func (i *Identify2WithUIDTester) CheckHint(h libkb.SigHint) libkb.ProofError {
+func (i *Identify2WithUIDTester) CheckHint(_ *libkb.GlobalContext, h libkb.SigHint) libkb.ProofError {
```
Comment (oconnor663) [inline go/engine/identify2_test.go]: Is it weird that `i` has a global context member, but also receives this argument?

Comment (maxtaco) [inline go/engine/identify2_test.go]: A little bit, but it needs to have it to meet the contract...

Comment (maxtaco) [inline go/engine/identify2_test.go]: The other alternative is to ferret the global context away in the proof scraper classes, but we'll have to do a whole lot of plumbing there too, and it's slightly less convenient.

File: go/libkb/proof_support_rooter.go [hunk 2]
```diff
@@ -124,20 +124,20 @@ func (rc *RooterChecker) rewriteURL(s string) (string, error) {
 	return u3.String(), nil
 }
 
-func (rc *RooterChecker) CheckStatus(h SigHint) (perr ProofError) {
+func (rc *RooterChecker) CheckStatus(g *GlobalContext, h SigHint) (perr ProofError) {
 
-	G.Log.Debug("+ Checking rooter at API=%s", h.apiURL)
+	g.Log.Debug("+ Checking rooter at API=%s", h.apiURL)
 	defer func() {
-		G.Log.Debug("- Rooter -> %v", perr)
```
Comment (oconnor663) [inline go/libkb/proof_support_rooter.go]: Yay!

**Comment by oconnor663** [comment]
:sheep:

**Pull request merged**